Content.Server/Fluids/EntitySystems/SpraySystem.cs

```diff
@@ -107,7 +106,8 @@ private void OnAfterInteract(EntityUid uid, SprayComponent component, AfterInter
 
             // impulse direction is defined in world-coordinates, not local coordinates
             var impulseDirection = Transform(vapor).WorldRotation.ToVec();
-            _vaporSystem.Start(vaporComponent, impulseDirection, component.SprayVelocity, target, component.SprayAliveTime);
```

Comment:
Vapor component is getting called a bunch so you should cache it as it isn't cheap.